java-extras/src/test/java/org/triplea/io/ZipExtractorTest.java

```diff
@@ -0,0 +1,66 @@
+package org.triplea.io;
+
+import static org.apache.commons.io.FileUtils.deleteDirectory;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.hamcrest.core.Is.is;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import org.junit.jupiter.api.Test;
+
+class ZipExtractorTest {
+
+  /**
+   * Unzips a sample test zip that contains the following files and file contents:
+   *
+   * <pre>
+   *   ./directory
+   *   ./directory/test-file2 : lupsom osculus
+   *   ./test-file1 : ipsum ispor
+   * </pre>
+   *
+   * Once the zip file is unzipped, the test verifies the files exist with expected contents.
+   */
+  @Test
+  void unzip() throws IOException {
+    final File destinationFolder = new File("destination");
+    destinationFolder.deleteOnExit();
+    try {
+      final File testDataZip = testDataZip();
+
+      ZipExtractor.unzipFile(testDataZip, destinationFolder);
+
+      assertThat(
+          "Should contain one folder and one file only",
+          FileUtils.listFiles(destinationFolder),
+          hasSize(2));
+
+      final File unzippedTestFile1 = new File(destinationFolder, "test-file1");
+      final String testFile1Contents = Files.readString(unzippedTestFile1.toPath());
+      assertThat(testFile1Contents, is("ipsum ipsor\n"));
+
+      final File unzippedDirectory = new File(destinationFolder, "directory");
+      assertThat(unzippedDirectory.isDirectory(), is(true));
+      assertThat(unzippedDirectory.exists(), is(true));
+
+      final File unzippedTestFile2 = new File(unzippedDirectory, "test-file2");
+      final String testFile2Contents = Files.readString(unzippedTestFile2.toPath());
+      assertThat(testFile2Contents, is("lupsom osculus\n"));
+    } finally {
+      deleteDirectory(destinationFolder);
```

Comment:
Only empty directories can be deleted, so it depends if Files.walk is depth first or breadth first. It does appear to be depth first. In this case I'm happy using a library to avoid the questions (just easier on maintainers). I agree what you posted should do the same thing.